resources/fetchers/ecr_factory.go

```diff
@@ -36,19 +38,15 @@ const (
 )
 
 func init() {
-	manager.Factories.ListFetcherFactory(ECRType, &ECRFactory{extraElements: getEcrExtraElements})
+	manager.Factories.ListFetcherFactory(ECRType, &ECRFactory{
+		KubernetesProvider: providers.KubernetesProvider{},
+		IdentityProvider:   awslib.GetIdentityClient,
+	})
 }
 
 type ECRFactory struct {
-	extraElements func() (ecrExtraElements, error)
-}
-
-type ecrExtraElements struct {
-	awsConfig               awslib.Config
-	kubernetesClientGetter  providers.KubernetesClientGetter
-	identityProviderGetter  awslib.IdentityProviderGetter
-	ecrPrivateRepoDescriber awslib.EcrRepositoryDescriber
-	ecrPublicRepoDescriber  awslib.EcrRepositoryDescriber
+	KubernetesProvider providers.KubernetesClientGetter
+	IdentityProvider   func(cfg awssdk.Config) awslib.IdentityProviderGetter
```

Comment:
wouldn't it be more straightforward by holding the `IdentityProviderGetter` here instead of holding a function to provide the provider?